test/learning/TestWeightUpdater.cpp

```diff
@@ -0,0 +1,99 @@
+#include "../../include/app.h"
+#include "../../include/learning/WeightUpdater.h"
+#include "../../include/types/AlternativesPerformance.h"
+#include "../../include/types/Categories.h"
+#include "../../include/types/Category.h"
+#include "../../include/types/Criteria.h"
+#include "../../include/types/MRSortModel.h"
+#include "../../include/types/Performance.h"
+#include "gtest/gtest.h"
+#include <sstream>
+#include <utility>
+
+Config getWeightTestConf() {
+  Config conf;
+  conf.data_dir = "../data/tests/";
+  try {
+    conf.logger =
+        spdlog::basic_logger_mt("test_logger", "../logs/test_logger.txt");
+  } catch (const spdlog::spdlog_ex &ex) {
+    conf.logger = spdlog::get("test_logger");
+  }
+  spdlog::set_level(spdlog::level::debug);
+  return conf;
+}
+
+AlternativesPerformance getAPTest() {
+  std::vector<Performance> perf_vect;
+  Criteria crits = Criteria(2, "crit");
+  std::vector<float> vec0 = {0.8, 0.6}; // good candidate
+  std::vector<float> vec1 = {1, 0.5};   // good candidate
+  std::vector<float> vec2 = {0.2, 0.4}; // bad candidate
+  perf_vect.push_back(Performance(crits, vec0, "a0"));
+  perf_vect.push_back(Performance(crits, vec1, "a1"));
+  perf_vect.push_back(Performance(crits, vec2, "a2"));
+  Category cat0 = Category("cat0", 0); // bad category
+  Category cat1 = Category("cat1", 1); // good category
+
+  std::unordered_map<std::string, Category> map =
```

Comment:
This will have to change will the disappearance of Performance.

> u will have to include utils in the test file
> change Performance with createVectorPerf("a0, cries, vec0)

Reply:
Will do after yours is merged!